crates/iceberg/src/spec/table_metadata_builder.rs

```diff
@@ -740,6 +740,32 @@ impl TableMetadataBuilder {
             .set_default_partition_spec(Self::LAST_ADDED)
     }
 
+    /// Remove partition specs by their ids from the table metadata.
+    /// Does nothing if a spec id is not present.
+    ///
+    /// If the default partition spec is removed, it is re-added
+    /// upon build.
+    pub fn remove_partition_specs(mut self, spec_ids: &[i32]) -> Self {
+        let mut removed_specs = Vec::with_capacity(spec_ids.len());
+
+        self.metadata.partition_specs.retain(|k, _| {
+            if spec_ids.contains(k) {
```

Comment:
Hi, `spec_ids.contains(k)` is an `O(n)` operation, whereas `partition_specs.get(x)` is `O(1)` (if I remember correctly that `partition_specs` uses a `HashMap`). Do you think it's a good idea to loop through `spec_ids` instead?

Reply:
Why not turn the `spec_ids` into a `HashSet`? :)

Reply:
> Why not turn the `spec_ids` into a `HashSet`? :)

I have thought about this, but it would introduce an additional allocation, which I believe isn't worth it.

Reply:
I switched the order - makes more sense to me this way. We are now looping over a probably very short array doing O(1) lookups. `partition_specs` is a HashMap